### PR TITLE
Datasource: twingate service accounts

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/223

## Changes
- added new datasource for twingate service accounts

### CODE REVIEW

File: docs/data-sources/service_accounts.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_service_accounts Data Source - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  Services offer a way to provide programmatic, centrally-controlled, and consistent access controls. For more information, see Twingate's documentation https://www.twingate.com/docs/services.
```
Comment (alexmensch) [inline docs/data-sources/service_accounts.md:6]: s/Services/Service Accounts/

Remove the sentence starting "For more information..."

File: docs/data-sources/service_accounts.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+# twingate_service_accounts (Data Source)
+
+Services offer a way to provide programmatic, centrally-controlled, and consistent access controls. For more information, see Twingate's [documentation](https://www.twingate.com/docs/services).
```
Comment (alexmensch) [inline docs/data-sources/service_accounts.md:11]: s/Services/Service Accounts/

File: docs/data-sources/service_accounts.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  name = "<your service account's name>"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `name` (String) Filter results by the name of the service account.
```
Comment (alexmensch) [inline docs/data-sources/service_accounts.md:26]: s/service account/Service Account/

File: docs/data-sources/service_accounts.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+- `services` (Block List) List of Services (see [below for nested schema](#nestedblock--services))
```
Comment (alexmensch) [inline docs/data-sources/service_accounts.md:31]: s/Services/Service Accounts/

Comment (alexmensch) [inline docs/data-sources/service_accounts.md:31]: We should call this value `service_accounts` for consistency

**Review by alexmensch — CHANGES_REQUESTED** [review]

File: docs/data-sources/service_accounts.md (hunk 1 above)
Comment (vmanilo) [reply docs/data-sources/service_accounts.md]: @alexmensch docs is updated

File: docs/data-sources/service_accounts.md (hunk 2 above)
Comment (vmanilo) [reply docs/data-sources/service_accounts.md]: updated

File: docs/data-sources/service_accounts.md (hunk 3 above)
Comment (vmanilo) [reply docs/data-sources/service_accounts.md]: updated

File: docs/data-sources/service_accounts.md (hunk 4 above)
Comment (vmanilo) [reply docs/data-sources/service_accounts.md]: updated

File: twingate/internal/model/service-account.go [hunk 5]
```diff
@@ -12,3 +12,19 @@ func (s ServiceAccount) GetName() string {
 func (s ServiceAccount) GetID() string {
 	return s.ID
 }
+
+type Service struct {
```
Comment (ekampf) [inline twingate/internal/model/service-account.go:16]: Remove

File: twingate/provider.go [hunk 6]
```diff
@@ -47,6 +47,7 @@ func Provider(version string) *schema.Provider {
 			datasource.TwingateConnectors:    datasource.Connectors(),
 			datasource.TwingateResource:      datasource.Resource(),
 			datasource.TwingateResources:     datasource.Resources(),
+			datasource.TwingateServices:      datasource.Services(),
```
Comment (ekampf) [inline twingate/provider.go:50]: ServiceAccounts

File: twingate/internal/provider/datasource/services.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/Twingate/terraform-provider-twingate/twingate/internal/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	fieldID        = "id"
+	fieldName      = "name"
+	fieldServices  = "service_accounts"
+	fieldResources = "resource_ids"
```
Comment (ekampf) [inline twingate/internal/provider/datasource/services.go:15]: fieldResourceIDs

File: twingate/internal/provider/datasource/services.go [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+	fieldKeys      = "key_ids"
```
Comment (ekampf) [inline twingate/internal/provider/datasource/services.go:16]: same

File: twingate/internal/client/service-account.go [hunk 9]
```diff
@@ -129,14 +140,153 @@ func (client *Client) readServiceAccountsAfter(ctx context.Context, variables ma
...
 
 	if len(response.Edges) == 0 {
-		return nil, NewAPIErrorWithID(ErrGraphqlResultIsEmpty, "read", serviceAccountResourceName, "All")
+		return nil, NewAPIErrorWithID(ErrGraphqlResultIsEmpty, operationRead, serviceAccountResourceName, "All")
 	}
 
 	return &response.PaginatedResource, nil
 }
+
+func (client *Client) ReadServices(ctx context.Context, input ...string) ([]*model.Service, error) {
```
Comment (ekampf) [inline twingate/internal/client/service-account.go:155]: ServiceAccounts

File: twingate/provider.go (hunk 6 above)
Comment (vmanilo) [reply twingate/provider.go]: fixed

File: twingate/internal/provider/datasource/services.go (hunk 8 above)
Comment (vmanilo) [reply twingate/internal/provider/datasource/services.go]: fixed

File: twingate/internal/client/service-account.go (hunk 9 above)
Comment (vmanilo) [reply twingate/internal/client/service-account.go]: fixed

**Review by alexmensch — APPROVED** [review]

**Review by ekampf — APPROVED** [review]

**Pull request merged**